ballista/executor/src/circuit_breaker/stream.rs

```diff
@@ -89,23 +107,9 @@ impl Stream for CircuitBreakerStream {
 
         let poll = self.inner.poll_next_unpin(cx);
 
-        if let Poll::Ready(Some(Ok(record_batch))) = &poll {
-            let delta = self.calculate.calculate_delta(record_batch);
-
-            self.percent += delta;
-
-            let status_update = CircuitBreakerUpdate {
-                key: self.key.clone(),
-                percent: self.percent,
-            };
-
-            if let Err(e) = self.client.send_update(status_update) {
-                if !self.is_lagging {
-                    self.is_lagging = true;
-                    warn!("Stream could not send short circuit update to daemon, it might be running very fast! ({:?})", e);
-                }
-            }
-        }
+        let delta = self.calculate.calculate_delta(&poll);
```

Comment:
With the rest of these changes we should be able to send updates only when required right? It will receive the stage tripped signal for this task for any update sent to the the scheduler.

Reply:
Okay, I've pushed another change, which I think should do what you're suggesting.

* Do not add update to update batch request if zero delta
* Send empty batch request anyways so we can receive the signals